Makefile

```diff
@@ -1,9 +1,25 @@
 lint:
 	golangci-lint run --config golangci.yaml
 
-pre-pr: lint
-	$(MAKE) -C ./example clean run
+.PHONY: pre-pr
+pre-pr: lint test-integration
 
 generate:
 	easyjson --all basm/dto.go
 	easyjson x/xbasm/parse.go
+
+sdk_srcs := $(wildcard **/*.go)
+wasm_src_dir := ./test/integration/testdata
+wasm_src := $(wasm_src_dir)/main.go
+wasm_out := $(wasm_src_dir)/x.wasm
+
+.PHONY: wasm
+wasm: $(wasm_out)
+
+$(wasm_out): $(wasm_src) $(sdk_srcs)
+	@echo "Building WASM module..."
+	@tinygo build -o $@ -target=wasi $<
+
+.PHONY: test-integration
+test-integration: $(wasm_out)
+	@go test -v ./test/integration/... -count=1
```

Comment:
These additions to the makefile are intended to manage the fact that the tests are dependent on the wasm binary. I think there is a possible future where there isn't just one wasm binary, but at the very least, if the wasm binary or its dependencies haven't changed, you should be able to run tests fairly quickly (no recompile). I figure this could serve as something to try out, and if we like this tooling setup we can use it as a basis for expanding the testing in the SDK if we want to add more.

Reply:
I added the `make wasm` target for convenience if folks are rebuilding a lot outside of running the tests.